x-pack/plugins/lens/public/xy_visualization/xy_expression.tsx

```diff
@@ -452,13 +467,20 @@ export function XYChart({
             yScaleType,
             xScaleType,
             isHistogram,
+            yConfig,
           } = layer;
           const columnToLabelMap: Record<string, string> = columnToLabel
             ? JSON.parse(columnToLabel)
             : {};
 
           const table = data.tables[layerId];
 
+          const shouldShowValueLabel =
+            !chartHasMoreThanOneSeries &&
+            shouldRotate &&
+            yConfig &&
+            yConfig?.find(({ forAccessor }) => forAccessor === accessor)?.showValueLabels;
```

Comment:
I think the "show value labels" option should not be set on any layer, but on the chart as a whole.